.github/workflows/try.yml

```diff
@@ -20,15 +28,34 @@ jobs:
         with:
           clean: False
 
-      - name: Clone qibo repository
-        uses: actions/checkout@v2
+      #- name: Clone qibo repository
+      #  uses: actions/checkout@v2
+      #  with:
+      #    repository: qibogang/qibo
+
+      - name: Deploy documentation
+        run: |
+          python -m pip install --upgrade pip
```

Comment:
Use instead [`setup-python`](https://github.com/actions/setup-python) action, it will do whatever is needed also concerning `pip`, no need to manually upgrade.

Explicitly specify Python version there, and if possible write it in a workflow variable (see [`inputs`](https://docs.github.com/en/actions/using-workflows/reusing-workflows#using-inputs-and-secrets-in-a-reusable-workflow)), such that we have it under control.

Reply:
You can also cache `pip` dependencies: it will be useless with Poetry, but for the time being we still have two projects Poetry-less.
https://github.com/actions/setup-python#caching-packages-dependencies

If you wish, you can already store the choice in a workflow variable (default to `pip`), and put the value here, such that you can already choose `poetry` if needed.